sampler_remote_test.go

```diff
@@ -599,3 +601,50 @@ func TestRemotelyControlledSampler_printErrorForBrokenUpstream(t *testing.T) {
 	sampler.UpdateSampler()
 	assert.Contains(t, logger.String(), "failed to fetch sampling strategy:")
 }
+
+func TestRemotelyControlledSampler_samplingStrategiesFile(t *testing.T) {
+	content := []byte("probabilistic")
```

Comment:
I am confused by this - a single word "probabilistic" is not a valid sampling strategy, how does this work?

Reply:
This was based on how other tests were doing it ¯\_(ツ)_/¯. I can take another look. Like I said previously, I can't find any tests which actually stressed the existing remote sampling logic.

Reply:
I wasn't getting coverage for `SamplerOptions.SamplingStrategiesFile` without it, because the other tests use the remote HTTP fetcher (eg: TestRemoteSamplerOptions).

Reply:
Can that part be tested in isolation? If you want to test it with whole Remote sampler, then the policy in the file needs to be (a) parseable, and (b) different from the default sampler, so that the test can assert end-to-end policy application.

Reply:
I'm working on using a full valid config. I actually am noticing that there may need to be client side logic to find the right section of the config for the provided serviceName. I'm looking at the upstream jaeger project to figure out how the collector/agent does this.

Reply:
There is a difference between config that is retrieved by SDK from the backend and the config that the backend has. The former is always scoped to the service making the request for sampling strategies, so the response would not contain a map for different services. Specifically, the API between remote sampler and the backend is the JSON version of this: https://github.com/jaegertracing/jaeger-idl/blob/34396033ff11c60fced342ab2858ace278fedaa8/thrift/sampling.thrift#L50